esp-hal/src/spi/master.rs

```diff
@@ -951,6 +951,13 @@ mod dma {
         Mode,
     };
 
+    fn empty_rx_buf() -> DmaRxBuf {
```

Comment:
Empty buffers should still have at least one descriptor in them and I want to keep this invariant.

Rather than change `DmaTxBuf` and `DmaRxBuf` to support zero descriptors, just introduce an `EmptyBuf` type who's single descriptor is a `#[ram] static EMPTY: DmaDescriptor = DmaDescriptor::EMPTY` (this could probably be `const`, not sure).

Or perhaps you can also just create a `DmaTxBuf` from this static descriptor.

Reply:
I guess this is a fair point.